python/lsst/ts/Dome/mock_llc_statuses/amcs_status.py

```diff
@@ -55,23 +54,34 @@ async def determine_status(self, current_tai):
         )
         if self.status != LlcStatus.STOPPED.value:
             azimuth_step = self.motion_velocity * time_diff
-            if self.motion_direction == motion_dir.CW.value:
-                self.position_actual = self.position_orig + azimuth_step
+            self.position_actual = self.position_orig + azimuth_step
+            if self.motion_velocity >= 0:
                 if self.position_actual >= self.position_cmd:
                     self.position_actual = self.position_cmd
-                    self.position_orig = self.position_actual
                     if self.status == LlcStatus.MOVING.value:
-                        self.status = LlcStatus.STOPPED.value
+                        if self.crawl_velocity >= 0:
+                            self.position_cmd = math.radians(720)
+                        else:
+                            self.position_cmd = math.radians(-360)
+                        self.motion_velocity = self.crawl_velocity
+                        self.status = LlcStatus.CRAWLING.value
                     else:
+                        self.position_cmd = self.position_actual
+                        self.motion_velocity = 0
                         self.status = LlcStatus.PARKED.value
             else:
```

Comment:
Prior code, but I find the logic for changing to PARKED hard to follow and hard to trust. It seems to be "if not STOPPED and not MOVING and arrived then PARKED". That may be fine, but what if some new states are added, such as HOMING and HOMED?

I think it would be more robust to only transition to PARKED if the prior state was PARKING, and if you run out of options then raise an exception.

Reply:
This evolved from the situation where there was no PARKING and PARKED yet and I agree that this could have been done in a better way. I followed your suggestion.